extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientAdditionalProviderBuildItem.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.restclient.deployment;
```

Comment:
Is it really necessary to replace the build item ? That item, at least in theory, can already be used to register some other custom providers via the annotation shortcuts

Reply:
I see you meant to reflect these are additional providers, but they are added via the annotation shortcuts...So just renaming the build item is probably not warranted

Reply:
It is not, I actually thought about this long and hard. Here are my thoughts:

- if the build item is used outside of Quarkus, it should be in SPI, otherwise we are going to be very restricted in the future in any improvements/refactoring etc if we can't remove/replace build items in deployment
- the build item is backwards compatible (I made sure they can use constructors accepts the same formal parameters), I only renamed it so that it's clear what the build item does

I don't mind renaming it back, please provide your feedback @sberyozkin

Reply:
> I see you meant to reflect these are additional providers, but they are added via the annotation shortcuts...So just renaming the build item is probably not warranted

That's why I added Javadoc to constructors. I don't know how to do this better (apart of finding even better name :-/). Any idea how I could improve it? Or should I just return original name?

Reply:
@sberyozkin on second thought, I'll simply provide a new build item, keep the original one and merge them in build step

Reply:
will do in a minute

Reply:
@michalvavrik IMHO it makes sense to keep the original name, I agree with your SPI comments, but for now, to keep the PR a bit simpler it makes sense to keep the original name, even if the possibility of it being reused is not high - I'm pretty sure though I was commenting somewhere on Zulip in relation to some queries about registering the providers from extensions etc... It is not really an additional provider what is being registered (i.e it is like any other provider available for the registration), it is a method of registering this provider which the original name (likely confusingly) tried to represent